src/olympia/addons/serializers.py

```diff
@@ -516,21 +519,24 @@ def _check_for_existing_versions(self, version_string):
     def validate(self, data):
         if not self.instance:
             guid = self.addon.guid if self.addon else self.parsed_data.get('guid')
-            self._check_blocklist(guid, self.parsed_data.get('version'))
+            version_string = self.parsed_data.get('version')
+            self._check_blocklist(guid, version_string)
             if self.addon:
-                self._check_for_existing_versions(self.parsed_data.get('version'))
-
-                # Also check for submitting listed versions when disabled.
-                if (
-                    data['upload'].channel == amo.CHANNEL_LISTED
-                    and self.addon.disabled_by_user
-                ):
-                    raise exceptions.ValidationError(
-                        gettext(
-                            'Listed versions cannot be submitted while add-on is '
-                            'disabled.'
+                self._check_for_existing_versions(version_string)
+
+                if data['upload'].channel == amo.CHANNEL_LISTED:
+                    if error_message := is_version_number_not_greater_than_current(
+                        self.addon, version_string
+                    ):
+                        raise exceptions.ValidationError({'version': error_message})
+                    if self.addon.disabled_by_user:
+                        # Also check for submitting listed versions when disabled.
```

Comment:
Nit: I think this comment makes sense above line 532 (in a similar spot to where it was before).